config/v1/types_infrastructure.go

```diff
@@ -505,9 +505,73 @@ type BareMetalPlatformStatus struct {
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
 }
 
+// BGPPeer describes a BGP peer.
+type BGPPeer struct {
+	// asn is the Autonomous System Number of the peer.
+	//
+	// +optional
+	ASN string `json:"asn,omitempty"`
```

Comment:
@pierreprinetti @mdbooth I wonder if we can remove this one, since we already have the ASN in the speaker configuration. I don't think a node would peer with more than one ASNs (not sure it's even possible)...